### PR TITLE
[geografir/geometry] add Geometry.to_crs

### PR DESCRIPTION
**Pull request by kissmygritts**
## Package(s)

`geografir/geometry`

## Description

Adds method and tests for `Geometry.to_crs`. I imagine this function may cause some weirdness as we hit more edge cases. For example, we aren't trying to wrap the dateline at all. This is slightly complicated, and has slightly unclean implementations (dateline = -180/180 degrees in geographic CRS). I'm currently thinking about how to implement this, but it is not currently a requirement for us.

I may add similar functionality to `BoundingBox`, haven't decided yet. It currently exists in `vp-airflow` but we don't have a good use case yet.

## Test plan

Tests pass in CI

### CODE REVIEW

File: geometry/src/geometry/geometry.py [hunk 1]
```diff
@@ -92,7 +97,36 @@ def __init__(self, geometry: BaseGeometry, crs: CRS | int | str):
...
+
+        # create transform object (this can fail if transforming between incompatible projects, this should be rare)
+        try:
+            transformer = Transformer.from_crs(self.crs, target_crs, always_xy=True)
+        except Exception as e:
+            raise TransformError(
+                f"Cannot create transformation from {self.crs} to {target_crs}"
+            ) from e
+
+        # then apply trasnform
```
Comment (jlw5) [inline geometry/src/geometry/geometry.py:125]:
```suggestion
        # then apply transform
```

**Review by jlw5 — APPROVED** [review]
💯
